teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -14,113 +14,138 @@ authors:
-schedule execution.
+As stated in Tekton's [reusability design principles](https://github.com/tektoncd/community/blob/main/design-principles.md#reusability),
+Pipelines and Tasks are meant to capture authoring-time concerns, and to be reusable in a variety of execution contexts.
+PipelineRuns and TaskRuns should be able to control execution without the need to modify the corresponding Pipeline or Task.
 
-## Motivation
+However, because each TaskRun is executed in a separate pod, Task and Pipeline authors indirectly control the number of pods used in execution.
+This introduces both the overhead of extra pods and friction associated with moving artifacts between Tasks.
 
-* Tasks are a reusable unit combining one or more steps (container images) together and defining an interface to those
-  steps via params, workspaces and results
-* Tasks can be combined in a Pipeline
-* When a Pipeline is invoked via a PipelineRun, each Task is executed as a separate pod.
+This TEP lists the pain points associated with running each TaskRun in its own pod and describes the current features that mitigate these pain points.
+It explores several options for decoupling Task composition and TaskRun scheduling but does not yet propose a preferred solution.
 
-This means that choices made around Task design (e.g. creating a Task that encapsulates a git clone and a separate Task
-to run go unit tests) directly impact the performance and overhead involved in executing the Tasks. For example
-if the git clone task wants to share data with the unit test task, beyond a simple
-[result](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-results),
-you'll need to provision a PVC or do some other similar, cloud specific storage,
-to [make a volume available](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#specifying-volumesources-in-workspaces)
-that can be shared between them, and running the second Task will be delayed by the overhead of scheduling a second pod.
+## Motivation
 
-### Goals
+The choice of one pod per Task works for most use cases for a single TaskRun, but can cause friction when TaskRuns are combined in PipelineRuns.
+These problems are exacerbated by complex Pipelines with large numbers of Tasks.
+There are two primary pain points associated with coupling each TaskRun to an individual pod: the overhead of each additional pod
+and the difficulty of passing artifacts between Tasks in a Pipeline.
+
+### Pod overhead
+
+Pipeline authors benefit when Tasks are made as self-contained as possible, but the more that Pipeline functionality is split between modular Tasks,
+the greater the number of pods used in a PipelineRun. Each pod consumes some system resources in addition to the resources needed to run each container
+and takes time to schedule. Therefore, each additional pod increases the latency of and resources consumed by a PipelineRun.
+
+### Difficulty of moving artifacts between Tasks
```

Comment:
kind of a nit but maybe important: i would say something like "data" instead of "artifacts" - i think what you said is accurate but to me artifact means specifically files you need to retain after the execution of the Pipeline has completed, usually for running software (e.g. images, binaries) but sometimes for providing history (e.g. test results)

but i think the pain here is if anything _especially_ about data that doesn't need to be persisted after the Pipeline finishes, e.g. data clone from a git repo. tho it also applies to artifacts (which id see as a kind of data), the pain point becomes especially clear when you have to create some kind of persistent storage even for data you don't need to persist post execution

Reply:
great point! I think the comment thread above is pretty similar on this so I'll resolve this to keep that discussion in one place.